db/migrations/deploy/20191211011147-fixindexperf.sql

```diff
@@ -0,0 +1,65 @@
+-- Deploy sat-api-pg:20191211011147-fixindexperf to pg
+
+BEGIN;
+
+CREATE OR REPLACE FUNCTION api.search(
+  bbox numeric[] default NULL,
+  intersects json default NULL,
+  include text[] default NULL,
+  andquery text default NULL,
+  sort text default 'ORDER BY c.datetime',
+  lim int default 50,
+  next text default '0'
+) RETURNS setof api.collectionitems AS $$
+DECLARE
+res_headers text;
+prefer text;
+intersects_geometry data.geometry;
+BEGIN
+  --  prefer := current_setting('request.header.prefer');
+IF bbox IS NOT NULL THEN
+  intersects_geometry = data.ST_MakeEnvelope(
+    bbox[1],
+    bbox[2],
+    bbox[3],
+    bbox[4],
+    4326
+  );
+ELSIF intersects IS NOT NULL THEN
+  intersects_geometry = data.st_SetSRID(data.ST_GeomFromGeoJSON(intersects), 4326);
+END IF;
+RETURN QUERY EXECUTE
+FORMAT(
+  'SELECT
+  collectionproperties,
+  collection,
+  id,
+  c.geom,
+  c.bbox,
+  type,
+  assets,
+  geometry,
+  CASE WHEN $2 IS NULL THEN properties
+  ELSE (
+    SELECT jsonb_object_agg(e.key, e.value)
+    FROM jsonb_each(properties) e
+    WHERE e.key = ANY ($2)
+  )
+END as properties,
+datetime,
+links,
+stac_version
+FROM api.collectionitems c
+WHERE (
+  $1 IS NULL OR
```

Comment:
This query will never end up using an intersects json. $1 needs to be changed to "intersects_geometry IS NULL"

Reply:
If you look more carefully there is a USING block at https://github.com/developmentseed/sat-api-pg/pull/35/files#diff-c15feaf8ffaf7015c3e1ab5758b36805R58 which handles this substitution.

Reply:
d'oh. sorry, missed that.